### ISSUE TITLE
namespace + extends not working

### ISSUE THREAD

**Issue by NexZhu** [fabien0102/ts-to-zod#105]
## Bug description

When using `extends` inside `namespace`, there's an error:

```
// Without namespace, this works:
export interface BaseObject {
  "id": string;
}
export interface Object extends BaseObject {
  "name": string;
}

// With namespace, this will fail:
export namespace std {
  export interface BaseObject {
    "id": string;
  }
  export interface Object extends BaseObject {
    "name": string;
  }
}
```

Error with namespace:
```
 ›   Warning: Some schemas can't be generated due to circular dependencies:
 ›   stdObjectSchema
✔ Validating generated types
🎉 Zod schemas generated!
```

Only `BaseObject` get generated:
```ts
// Generated by ts-to-zod
import { z } from 'zod'

export const stdBaseObjectSchema = z.object({
  id: z.string(),
})
```

Without namepsace everything's good:
```ts
// Generated by ts-to-zod
import { z } from 'zod'

export const baseObjectSchema = z.object({
  id: z.string(),
})

export const objectSchema = baseObjectSchema.extend({
  name: z.string(),
})
```

**Versions**

ts-to-zod/1.13.1 wsl-x64 node-v19.3.0

**Comment by fabien0102** [comment]
Good catch, I'm betting than the extended object is missing the namespace prefix.

**Comment by NexZhu** [comment]
@fabien0102 There are still two likely related issues:

First one: Validation fails even though the output with `--skipValidation` looks correct.

Example input:
```ts
/* eslint-disable @typescript-eslint/no-namespace */
export namespace std {
  export interface BaseObject {
    id: string;
  }
}

export interface Object extends std.BaseObject {
  createdAt: Date;
}

```

Error without `--skipValidation`:
```
✖ Validating generated types
 ›   Error: 'Object' is not compatible with 'objectSchema':
 ›   Argument of type 'Object' is not assignable to parameter of type '{ id: 
 ›   string; createdAt: Date; }'.
 ›     Property 'id' is missing in type 'Object' but required in type '{ id: 
 ›   string; createdAt: Date; }'.
```

Output with `--skipValidation`, seems correct:
```ts
// Generated by ts-to-zod
import { z } from "zod";

export const stdBaseObjectSchema = z.object({
  id: z.string(),
});

export const objectSchema = stdBaseObjectSchema.extend({
  createdAt: z.date(),
});
```

**Comment by NexZhu** [comment]
Second one: 

Type alias syntax like this failed to generate (only if they are in a namespace):
```ts
/* eslint-disable @typescript-eslint/no-namespace */
export namespace std {
  export interface BaseObject {
    id: string;
  }
  export type Object = BaseObject;
}
```

Error:
```
 ›   Warning: Some schemas can't be generated due to circular dependencies:
 ›   stdObjectSchema
✔ Validating generated types
🎉 Zod schemas generated!
```

Same types without namespace works as expected:
```
export interface BaseObject {
  id: string;
}
export type Object = BaseObject;
```

**Comment by fabien0102** [comment]
Interesting, I will have a look

**Comment by decoded4620** [comment]
@fabien0102  I'm having a similar issue, with seemingly very simple types which use extends / implements, within a single `types` file. 
e.g. 
```
export interface BaseObject {...}

export interface ObjA extends BaseObject {
  fieldA: string;
}

export interface ObjB extends BaseObject {
 fieldB: number;
}
```

I don't want to pass things like `--skipValidation` but I feel this should work and produce a `schemas.ts` file like:

```
baseObjectSchema = z.object({});

objASchema = baseObjectSchema.extend({
   fieldA: z.string()
});

objBSchema = baseObjectSchema.extend({
   fieldB: z.number()
});
```

What are the rules here, and why isn't this a normal use case?

**Comment by fabien0102** [comment]
I will have a look today into this, something is broken in the validation ^^ The idea of this validator is to compare generated types `z.infer<>` with the original one and see if they are the same.
In the case of a namespace, we are doing a bit of transformation (adding a prefix since namespace is not a thing in zod), and something is not following during this type comparison.

Anyway, nothing to worry about, this validator is there just to help me find unhandled usecases, for this case, this is a false positive, you can skip it without be worried

**Comment by fabien0102** [comment]
I just tried on the latest version, the given repro step is now working (probably fixed by #114)
